arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedColumnIterator.java

```diff
@@ -69,12 +69,14 @@ public boolean producesDictionaryEncodedVector() {
   }
 
   public abstract class BatchReader {
-    public void nextBatch(FieldVector fieldVector, int typeWidth, NullabilityHolder holder) {
+    public void nextBatch(
+        int numValsToRead, FieldVector fieldVector, int typeWidth, NullabilityHolder holder) {
```

Comment:
Not sure how likely this is, but is "int" safe here? With compression we could have a vector that theoretically represents over INT.MAX rows. Maybe that's not a real concern though

Reply:
ah rowsReadSoFar is an Int so this doesn't matter.